src/python/pants/engine/target.py

```diff
@@ -719,32 +736,28 @@ class NoCacheField(BoolField):
 COMMON_TARGET_FIELDS = (Tags, DescriptionField, NoCacheField)
 
 
-# TODO: figure out what support looks like for this. This already gets hydrated into a
-#  List[Address] by the time we have a HydratedStruct. Should it stay that way, or should it go
-#  back to being a List[str] and we only hydrate into Addresses when necessary? Alternatively, does
-#  hydration mean getting back `Targets`?
-class Dependencies(AsyncField):
+# NB: We don't make this an `AsyncField` like you may expect because Struct and Addressable will
```

Comment:
By inspection the reader should never expect this Field to have needed to be an AsyncField IIUC since an Address is able to be determined from the string given on the CLI immediately with nothing except string parsing.

Reply:
Good point. I'll fix this comment.